Makefile

```diff
@@ -17,28 +17,49 @@
 #
 # Makefile with some common workflow for dev, build and test
 #
+.DEFAULT_GOAL:=help
 
-all: build test
+##@ General
 
-.PHONY: build test
+.PHONY: help
+help:  ## Display this help
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_0-9-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
 
-build:
+all: build install test ## Perform the build, install and run the go tests
+
+##@ Build
+
+.PHONY: build
+build: ## Build the kubebuilder
 	go build -o bin/kubebuilder ./cmd
 
-install: build
+install: build ## Install a binary locally with the current source code. Use it to test your changes.
 	cp ./bin/kubebuilder $(shell go env GOPATH)/bin/kubebuilder
 
-generate:
+##@ Generation
+
+generate-all: generate ## Update/generate all mock data
```

Comment:
@mengqiy,

What is exactly the oder to generate all mock data locally? What are the pre-requirements and currently workarrounds?